### ISSUE TITLE
Craft 4?

### ISSUE THREAD

**Issue by angieherrera** [besteadfast/craft-preparse-field#80]
Hey guys! Will this work with Craft 4 or do we need to wait for an update? Thanks!

**Comment by jalendport** [comment]
Hey! The current plan is to get it Craft 4 ready in the next week or so and release as v1.3, and then in the next couple months I'm looking into rewriting the whole thing from the ground up (to reduce bugginess and give us a better foundation for the plugin moving forward) and eventually releasing that as v2.

**Comment by howells** [comment]
@jalendport sorry to chase! Just wondering if there was any target date for Craft 4?

**Comment by TUSAB-web** [comment]
I also wanted to check on the Craft 4 compatibility timeline.

**Comment by DTesch-Reem** [comment]
Any update regarding the Craft 4 upgrade?

**Comment by jcdarwin** [comment]
For anyone who can't wait for the maintainer of this repo to upgrade it to Craft 4, I've done so at: 
https://github.com/nzmebooks/craft-preparse-field/tree/master

There's a corresponding PR at: https://github.com/besteadfast/craft-preparse-field/pull/84

**Comment by Havnevik** [comment]
> For anyone who can't wait for the maintainer of this repo to upgrade it to Craft 4, I've done so at: https://github.com/nzmebooks/craft-preparse-field/tree/master
> 
> There's a corresponding PR at: #84

Hi @jcdarwin, I have never used a fork before in any of my projects so please excuse my ignorance.

I have this in my main composer.json file:

"repositories": [
      {
        "type": "vcs",
        "url": "https://github.com/nzmebooks/craft-preparse-field"
      }
    ],
    "require": {
        "besteadfast/craft-preparse-field": "dev-develop",
        "craftcms/cms": "^4.0.0"
}

But I get this when I try composer update :

Problem 1
    - Root composer.json requires besteadfast/craft-preparse-field dev-develop -> satisfiable by besteadfast/craft-preparse-field[dev-develop].
    - besteadfast/craft-preparse-field dev-develop requires craftcms/cms ^3.2.0 -> found craftcms/cms[3.2.0-alpha.1, ..., v3.x-dev] but it conflicts with your root composer.json require (^4.0.0).

and when I try composer require I get:


In PackageDiscoveryTrait.php line 379:
                                                                                                                                                                                                                             
  Could not find a matching version of package nzmebooks/craft-preparse-field. Check the package spelling, your version constraint and that the package is available in a stability which matches your minimum-stability (d  
  ev). 


What I am doing wrong?

**Comment by mmikkel** [comment]
@Havnevik The `repositories` entry in your `composer.json` should look like this:

```json
"repositories": {
    "besteadfast/craft-preparse-field": {
        "type": "vcs",
        "url": "https://github.com/nzmebooks/craft-preparse-field"
    }
}
```

And it looks like @jcdarwin has tagged his fork as `2.0`, so to install it you need to do `composer require besteadfast/craft-preparse-field:^2.0`

**Comment by jalendport** [comment]
Hey everyone, thanks for your patience. I just released v2.0.0 which adds compatibility with Craft 4 🎉
